src/assets/custom/js/websiteHeader.test.js

```diff
@@ -1,120 +1,139 @@
-            expect(header.classList).toContain("website-header");
-            expect(header.classList.length).toBe(1);
+          afterAll(() => {
+            header.classList.remove(HEADER_HAS_OPEN_SUBMENU_CLASS);
           });
-        });
+          describe('And we scroll down the page by three pixels', () => {
+            beforeEach(() => {
+              simulateScrollingToY(1000);
+            });
 
-        describe("When we then scroll down the page by a couple of pixels", () => {
-          beforeEach(() => {
-            simulateScrollingToY(1000 - 1);
-          });
+            it('The submenu closes', () => {
+              expect(header.classList).not.toContain(
+                HEADER_HAS_OPEN_SUBMENU_CLASS
+              );
+            });
 
-          it("we don't immediately hide the header", () => {
-            expect(header.classList).toContain("website-header");
-            expect(header.classList).toContain("website-header--revealed");
-            expect(header.classList.length).toBe(2);
+            it('The header is hidden', () => {
+              expect(header.classList).not.toContain(
+                'website-header--revealed'
+              );
+            });
           });
         });
 
-        describe("When a submenu is open", () => {
-          beforeAll(() => {
-            header.classList.add(HEADER_HAS_OPEN_SUBMENU_CLASS);
-          });
-
-          describe("And we scroll down the page by three pixels", () => {
+        describe('when a submenu is not open', () => {
+          describe('And we scroll down the page by three pixels', () => {
             beforeEach(() => {
-              simulateScrollingToY(1000);
+              simulateScrollingToY(1000 + 100);
             });
 
-            it("we still don't hide the header", () => {
-              expect(header.classList).toContain("website-header");
-              expect(header.classList).toContain("website-header--revealed");
-              expect(header.classList.length).toBe(3);
+            it('The header is hidden', () => {
+              expect(header.classList).not.toContain(
+                'website-header--revealed'
+              );
             });
           });
         });
+      });
+
+      describe('When we then scroll close to the top of the page', () => {
+        beforeEach(() => {
+          simulateScrollingToY(1);
+        });
 
-        describe("When a submenu is not open", () => {
+        it("we let the header return to it's natural position", () => {
+          expect(header.classList).toContain('website-header');
+          expect(header.classList).toContain('website-header--revealed');
+          expect(header.classList.length).toBe(2);
+        });
+
+        describe('When we then scroll down the page by a couple of pixels', () => {
           beforeEach(() => {
-            header.classList.remove(HEADER_HAS_OPEN_SUBMENU_CLASS);
+            simulateScrollingToY(3);
```

Comment:
This isn't really part of these changes, but the more times we look at these tests the more I recognise the benefit of having a `simulateScrollingBy` function to go with `simulateScrollingToY`

Something like:
```javascript
function simulateScrollingBy(diff) {
  window.scrollY = window.scrollY + diff;

  window.document.dispatchEvent(
    new Event('scroll', { bubbles: true, cancelable: true })
  );


  triggerMockRequestAnimationFrame();
}
```

Reply:
That's a good idea. I think it allows the tests to be read a bit better. It's been implemented here: https://github.com/codurance/site/pull/1940/commits/82ac868abefbcd8d773ec6fea2cd5d184a094e8c